src/wallet/db.cpp

```diff
@@ -435,7 +467,9 @@ CDB::CDB(CWalletDBWrapper& dbw, const char* pszMode, bool fFlushOnCloseIn) : pdb
             if (ret != 0) {
                 throw std::runtime_error(strprintf("CDB: Error %d, can't open database %s", ret, strFilename));
             }
-            CheckUniqueFileid(*env, strFilename, *pdb_temp);
+            for (auto& env : g_dbenvs) {
```

Comment:
Is this required? Somehow I had assumed they only conficted if they were both in the same env, but I dunno where I got that from.

Reply:
Did you forget to push? Noting this is the only way we handle hardlinks is useful, indeed.

Reply:
> Did you forget to push?

Oops, yeah. It was on the pr/wfile.9 tag I linked to below. Should be updated now. Also I'll work on the "just leave this part of the diff" change you requested above.